src/lib/libdylink.js

```diff
@@ -1024,6 +1080,18 @@ var LibraryDylink = {
         }
       }
 
+#if FILESYSTEM && SUPPORT_RPATH
+      if (foundFile) {
+        var libData = FS.readFile(libName, {encoding: 'binary'});
+        if (libData) {
+#if DYLINK_DEBUG
+          dbg(`loaded library from filesystem: ${libName}`);
+#endif
+          return flags.loadAsync ? Promise.resolve(libData) : libData;
+        }
+      }
```

Comment:
It seems a little odd to re-introduce this alternative method of loading libraries.

Also, this check for `foundFile` looks like its inside the `if (dso) ` block, but the settings of `foundFile` is outside?